### PR TITLE
Launcher

### PR DESCRIPTION
**Pull request by pfh28**
Change icon and make a splash screen

### CODE REVIEW

File: app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/LoadingActivity.java [hunk 1]
```diff
@@ -1,98 +0,0 @@
...
-     * really it should be on a timer.
-     */
-    public void finishLoading(View view) {
-        Intent intent = new Intent(this, WelcomeActivity.class);
-        startActivity(intent);
-    }
-
-    private void initDatabase() {
-        // Initialize the enum tables. ====================================
-        try
```
Comment (jdy27) [inline app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/LoadingActivity.java]: What is in this try/catch is needed as these are tables that have to be populated in order to save workouts/transitions.  If you just add this function (minus the workout part) into the splash screen, that should fix our issue.

File: app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/SplashActivity.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+public class SplashActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
```
Comment (jdy27) [inline app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/SplashActivity.java]: Insert the initdatabase call here.

File: app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/LoadingActivity.java (hunk 1 above)
Comment (ptrgags) [reply app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/LoadingActivity.java]: I copied this code into the `SplashActivity`

File: app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/SplashActivity.java (hunk 2 above)
Comment (ptrgags) [reply app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/SplashActivity.java]: Done.

**Comment by ptrgags** [comment]
I fixed the things Jon pointed out. Someone should double check that things work properly.

**Comment by ptrgags** [comment]
I forgot to mention, make sure to Gradle sync before trying this branch, sometimes the IDE gets confused when you get rid of a layout all of a sudden

**Comment by btmosher1** [comment]
I tried testing this and it crashes on startup for me every time. I made sure to Gradle sync, but I cant get to the splash screen before crashing

File: app/src/main/res/drawable/splash.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item
+        android:drawable="@color/colorPrimary"/>
+
+    <item>
+        <bitmap
+            android:gravity="center"
+            android:src="@mipmap/ic_launcher"/>
```
Comment (jdy27) [inline app/src/main/res/drawable/splash.xml]: Looking into the current issue and here is the offending line.  One way I know of fixing it is by putting an ic_launcher image with the drawable folder, and reference that with @drawable/ic_launcher.  I was trying to look into the mipmap, but it seems like there are not a lot of resources on it.

**Comment by rkr35** [comment]
So I wasn't able to reproduce the crash—the app boots with the splash screen and DB init.

I'm assuming you guys already tried clearing the app data from Android Settings?
What does the Run log show in Android Studio? Any particular exception?

**Comment by jdy27** [comment]
> So I wasn't able to reproduce the crash—the app boots with the splash screen and DB init.
> 
> I'm assuming you guys already tried clearing the app data from Android Settings?
> What does the Run log show in Android Studio? Any particular exception?

I have tried cleaning the app data, as well as uninstalling, reinstalling and doing various gradle/file syncs and nothing has seemed to fix it.

There is a particular exception that I get every time: Caused by: android.content.res.Resources$NotFoundException: Drawable edu.drexel.lapcounter.lapcounter:drawable/splash with resource ID #0x7f080080
    Caused by: android.content.res.Resources$NotFoundException: File res/drawable/splash.xml from drawable resource ID #0x7f080080.

From looking into it, it seems to be related to the fact that the mipmap isnt in drawable or that its not refering to a drawable, but it does not explain how it would work on some devices but not others.

**Comment by btmosher1** [comment]
Tried this branch again, I've tried a many combinations of uninstalling, clearing data, and syncing files before running the branch, but still always crash. Heres the exception from my run log:

image

**Comment by ptrgags** [comment]
whoever's working on this branch, here's some ideas: 
1. try going through the XML files and search for every place where "splash" appears. make sure the paths to images are valid
2. Try re-adding the icon from scratch. see if this is any use: https://developer.android.com/studio/write/image-asset-studio

**Comment by rkr35** [comment]
Shout-out to @jdy27 for a simple fix to a crazy problem!

Merging.

**Pull request merged**